app/api/items/items.go

```diff
@@ -134,17 +134,17 @@ func insertItemItems(store *database.DataStore, spec []*insertItemItemsSpec) {
 	var values = make([]interface{}, 0, len(spec)*9)
 
 	for index := range spec {
-		values = append(values, store.NewID(),
+		values = append(values,
```

Comment:
The fact that this function is still called in the `RetryOnDuplicatePrimaryKeyError` in `add_item` (and probably others) cannot be a problem? In case there is a duplicate child, could it fail the same way in "loop" ?
It would be nice to prevent this issue not by checking manually there is no duplicate or by expecting the context to avoid such a case, but by failing properly on the DuplicatePrimaryKeyError on the `items_items` table (while keeping it on the `items` table)

(+ I assume the related services do test this scenario anyway)

Reply:
Fixed (although it could not cause problems)

Reply:
Why couldn't it cause problems?

Reply:
Oh, sorry. Yes, it really could cause problems because we still have the primary key in items_items (a composite one).

Reply:
Wouldn't there be a way to make sure in `RetryOnDuplicatePrimaryKeyError` cannot go wrong in general. In addition to the sql error code, do you get a message? If so, you could match the PK id name (that you would give when calling the fct) maybe?

Reply:
PK's name is always 'PRIMARY' in MySQL.

Reply:
> Wouldn't there be a way to make sure in RetryOnDuplicatePrimaryKeyError cannot go wrong in general.